src/components/segmentedControl/segmentedControl.svelte

```diff
@@ -0,0 +1,222 @@
+
+  type Size = 'default' | 'small' | 'tiny'
+
+  export let selected: string | undefined = undefined
+  export let size: Size = 'default'
+
+  let segmentedControl: HTMLDivElement;
+
+  let pillWidth: number;
+  let pillPosition: number;
+
+  $: console.log(selected)
+
+  const dispatch = createEventDispatcher<{
+    'select-item': any
+  }>()
+
+  function getValue(e: Element) {
+    // If the option element doesn't have a value, fallback to using the text
+    // content - this allows writing simplified options:
+    // i.e. <o>1</o>
+    return e.getAttribute('value') ?? e['value'] ?? e.textContent
+  }
+
+  $: controlItems = Array.from(
+    // TODO(petemil): support slot *descendents* that are leo-menu-item or leo-option (or something else unstyled like leo-menu-action?) so that we can
+    // select items that are in complex positions (see an example in the browser app menu zoom controls).
+    (segmentedControl?.querySelector('slot') as HTMLSlotElement)
+      ?.assignedElements()
+      ?.filter((element) =>
+        ['LEO-SEGMENTED-ITEM'].includes(element.tagName)
+      ) ??
+      segmentedControl?.querySelectorAll(
+        ':is(leo-segmented-item)'
+      ) ??
+      []
+  ) as HTMLElement[]
+
+  $: {
+    for (const controlItem of controlItems) {
+      controlItem.setAttribute('tabindex', '0')
+
+      if (controlItem.tagName === 'LEO-SEGMENTED-ITEM') {
+        controlItem.setAttribute('role', 'option')
+
+        if (selected === getValue(controlItem)) {
+          controlItem.setAttribute('aria-selected', '')
+          pillWidth = controlItem.clientWidth;
+          pillPosition = controlItem.offsetLeft;
+        } else controlItem.removeAttribute('aria-selected')
+      } else {
+        controlItem.setAttribute('role', 'menuitem')
+      }
+    }
+  }
+
+  function selectItem(e: Event) {
+    // Find the option which was clicked on, if any.
+    const item = controlItems.find((item) => e.composedPath().includes(item))
+
+    // If the event was triggered for something which isn't an option don't fire
+    // a change event.
+    if (!item) return
+
+    // Use data-is-interactive=true to prevent the menu from closing when selected. This implies
+    // there is interacitivity inside the menu item (e.g. a Toggle), which would be good for the user
+    // to see change state and allowing the user to manually close when ready. In other words, when we want
+    // the styles and menu navigation features of leo-menu-item, but we don't want the auto-close.
+
+    if (item.tagName === 'LEO-SEGMENTED-ITEM') {
+      selected = getValue(item)
+
+      dispatch('select-item', {
+        value: selected
+      })
+    }
+
+    // When using keyboard navigation to call selectItem, ensure click handlers happen on the item,
+    // but not when this was called in a click handler, otherwise we'll get 2x clicks.
+    if (item.tagName === 'LEO-SEGMENTED-ITEM' && e.type !== 'click') {
+      item.click()
```

Comment:
I'm having a bit of a brain fart, I can't remember why we do this. From memory, selecting something (like a button) with the keyboard actually triggers a click event....

Reply:
That's probably from the dropdown, now I'm thinking about it, so this control might be fine without it.